agent/protocol/nats/nats.go

```diff
@@ -0,0 +1,771 @@
+	if natsParser == nil {
+		common.ProtocolParserLog.Debugf("NATSStreamParser unsuport method[%v]", method)
+		index_ := readLine(buffer)
+		if index_ < 0 {
+			return protocol.ParseResult{ParseState: protocol.NeedsMoreData}
+		} else {
+			return protocol.ParseResult{ParseState: protocol.Ignore, ReadBytes: index_}
+		}
+	}
+	offset := index - len(method)
+	msg, prasedLen := natsParser.Parse(buffer[offset:], messageType)
+	if prasedLen < 0 {
+		return protocol.ParseResult{ParseState: protocol.NeedsMoreData}
+	}
+	if msg == nil {
+		return protocol.ParseResult{ParseState: protocol.Ignore, ReadBytes: prasedLen + offset}
+	}
+	return msg.packFrameBase(streamBuffer, prasedLen+offset)
+}
+
+func readField(buffer []byte) int {
+	return bytes.IndexFunc(buffer, func(b rune) bool { return b == ' ' || b == '\t' || b == '\r' || b == '\n' })
+}
+
+func readLine(buffer []byte) int {
+	index := bytes.Index(buffer, []byte(_CRLF_))
+	if index >= 0 {
+		index += len(_CRLF_)
+	}
+	return index
+}
+
+func splitFields(buffer []byte) [][]byte {
+	return bytes.FieldsFunc(buffer, func(r rune) bool { return r == ' ' || r == '\t' })
+}
+
+func trimFiled(buffer []byte) []byte {
+	return bytes.TrimFunc(buffer, func(r rune) bool { return r == ' ' || r == '\t' })
+}
+
+func (msg *NatsMessage) FormatToSummaryString() string {
+	return fmt.Sprintf("[protocol=[%v], subject=[%v]", msg.ProtocolCode.String(), msg.Subject)
+}
+
+func (msg *NatsMessage) FormatToString() string {
+	return fmt.Sprintf("protocol=[%v], subject=[%v], detail=[%s]", msg.ProtocolCode.String(), msg.Subject, msg.Buf)
+}
+
+func (msg *NatsMessage) IsReq() bool {
+	return msg.isReq
+}
+
+func (msg *NatsMessage) StreamId() protocol.StreamId {
+	if msg.ProtocolCode == PING || msg.ProtocolCode == PONG {
+		return 1
+	}
+	return 0
+}
+
+func (msg *NatsMessage) packFrameBase(streamBuffer *buffer.StreamBuffer, readBytes int) protocol.ParseResult {
+	fb, ok := protocol.CreateFrameBase(streamBuffer, readBytes)
+	if !ok {
+		common.ProtocolParserLog.Debugf("NATS Failed to create FrameBase for frameSize=%d", readBytes)
+		return protocol.ParseResult{ParseState: protocol.Ignore, ReadBytes: readBytes}
+	}
+	msg.FrameBase = fb
+	return protocol.ParseResult{
+		ParseState:     protocol.Success,
+		ParsedMessages: []protocol.ParsedMessage{msg},
+		ReadBytes:      readBytes,
+	}
+}
+
+// Info
+func (m *Info) String() string {
+	return fmt.Sprintf("Protocol:INFO,ServerID:%v,ServerName:%v,Version:%v,GoVersion:%v,Host:%v,Port:%v,MaxPayload:%v,TLSRequired:%v",
+		m.ServerID, m.ServerName, m.Version, m.GoVersion, m.Host, m.Port, m.MaxPayload, m.TLSRequired)
+}
+func (m *Info) ParseData(payload []byte) (*Info, int) {
+	// INFO {"option_name":option_value,...}␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) != 2 {
+		return nil, packetLen
+	}
+	// if strings.ToUpper(string(parts[0])) != "INFO" {
+	// 	return nil, packetLen
+	// }
```

Comment:
Commented code is always suspicious.

Either it could have been removed, or something was temporarily commented for debug and it was left commented while it shouldn't